src/python/foglamp/device/server.py

```diff
@@ -87,45 +88,45 @@ class Server:
         error = None
 
         try:
+            # Create Category before loading the plugin
             config = {}
             storage = Storage(cls._core_management_host, cls._core_management_port, svc=None)
             cfg_manager = ConfigurationManager(storage)
-            await cfg_manager.create_category(cls._plugin_name, config,
-                                                        '{} Device'.format(cls._plugin_name), True)
-
+            await cfg_manager.create_category(cls._plugin_name, config, '{} Device'.format(cls._plugin_name), True)
             config = await cfg_manager.get_category_all_items(cls._plugin_name)
 
-            try:
-                plugin_module_name = config['plugin']['value']
-            except KeyError:
-                _LOGGER.warning("Unable to obtain configuration of module for plugin {}".format(cls._plugin_name))
-                raise
+            # TODO: Examine if below code is as per Architecture
+            # try:
+            #     plugin_module_name = config['plugin']['value']
+            # except KeyError:
+            #     _LOGGER.warning("Unable to obtain configuration of module for plugin {}".format(cls._plugin_name))
+            #     raise
 
+            # Retrieve the plugin name and hence plugin module name, from the params
+            plugin_module_name = cls._plugin_name
```

Comment:
Why has this changed? Work was done in the previous sprint to pick up the correct plugin name, this appears to be undoing it.

Reply:
@MarkRiddoch please note that previous code required that the plugin config pre-existed in the configration table, which is contrary to the Architecture specs. Hence I have restored the commented lines AND have moved these lines in the except block. In this way, the code will work first time when there is no config in the db, via except block, and will work second time onwards via the main try block when the config is already available in db. Hope this is ok.

Reply:
@MarkRiddoch related to this is another query. How a plugin is supposed to supply its config info to DB the very first time the system is started? This line serves that purpose only. Second time onwards, this line will never be executed.

Reply:
@MarkRiddoch related to this is another query. How a plugin is supposed to supply its config info to DB the very first time the system is started? This line serves that purpose only. Second time onwards, this line will never be executed.